packages/pages/src/components/address/methods.ts

```diff
@@ -22,3 +27,152 @@ export const getUnabbreviated = (
+    address?.postalCode,
+    address?.countryCode,
+  ]);
+
+  switch (provider) {
+    case "APPLE":
+      return getDirectionsApple(query, directions);
+      break;
+    case "BING":
+      query = encodeArray([
+        address?.line1,
+        address?.city,
+        address?.region,
+        address?.postalCode,
+      ]);
+      return getDirectionsBing(query, directions);
+      break;
+    default:
+      const gmbURL = getlistingUrl(
+        profile.ref_listings,
+        ListingPublisher.googlemybusiness
+      );
+      if (gmbURL) {
+        return gmbURL;
+      }
+
+      if (profile.googlePlaceId) {
+        return getDirectionsGooglePlaceID(
+          profile.googlePlaceId,
+          query,
+          directions
+        );
+      }
+
+      return getDirectionsGoogle(query, directions);
+  }
+};
+
+/**
+ * Get Apple Maps location query
+ *
+ * @param {string} query - Stringified address query
+ * @param {boolean} directions
+ *
+ * @returns {string} - Apple maps url
+ */
+const getDirectionsApple = (query: string, directions?: boolean): string => {
+  return directions
+    ? `https://maps.apple.com/?daddr=${query}`
+    : `https://maps.apple.com/?address=${query}`;
+};
+
+/**
+ * Get Bing Maps location query
+ *
+ * @param {string} query - Stringified address query
+ * @param {string} directions
+ *
+ * @returns {string} - Bing maps url
+ */
+const getDirectionsBing = (query: string, directions?: boolean): string => {
+  return directions
+    ? `https://bing.com/maps/default.aspx?rtp=adr.${query}`
+    : `https://bing.com/maps/default.aspx?where1=${query}`;
+};
+
+/**
+ * Get a Google Maps Place ID page
+ *
+ * @param {string} placeId - Stringified address query
+ * @param {string} query - Stringified address query
+ * @param {boolean} directions - Enable driving directions
+ * @returns {string} - Google maps url
+ */
+const getDirectionsGooglePlaceID = (
+  placeId: string,
+  query: string,
+  directions?: boolean
+): string => {
+  return directions
+    ? `https://maps.google.com/maps/dir/?api=1&destination_place_id=${placeId}&destination=direct`
+    : `https://maps.google.com/maps/search/?api=1&query=${query}&query_place_id=${placeId}`;
+};
+
+/**
+ * Get a Google Maps search query
+ *
+ * @param {string} query - Stringified address query
+ * @param {boolean} directions - Enable driving directions
+ * @returns {string} - Google maps url
+ */
+const getDirectionsGoogle = (query: string, directions?: boolean): string => {
+  return directions
+    ? `https://maps.google.com/maps/dir/?api=1&destination=${query}`
+    : `https://maps.google.com/maps/search/?api=1&query=${query}`;
+};
+
+/**
+ * Get a maps url from listings
+ *
+ * @param {Yext.ListingType[]} listings - Yext listings
+ * @param {ListingPublisher} publisher - provider to get listing for
+ * @returns {string} - url
+ */
+const getlistingUrl = (
+  listings: ListingType[] = [],
+  publisher: ListingPublisher
+): string | undefined => {
```

Comment:
Based on my previous comment you shouldn't need this to return undefined.

Reply:
It isn't guaranteed that a client will have GMB setup. It's rare but you could buy pages without listings. Does this not assuming sound good?